src/beat_number.rs

```diff
@@ -1,90 +1,79 @@
-#[derive(PartialEq, Debug)]
+#[derive(PartialEq, Debug, Clone, Copy)]
 pub struct BeatNumber {
-    pub quarter_note: u32,
```

Comment:
I saw your comment on the previous PR, definitely simplifying to just represent this as the single sixteenth note field. While I was at it, I changed it to be zero-based